src/lamp_py/ingestion/glides.py

```diff
@@ -0,0 +1,405 @@
+
+    @property
+    def event_schema(self) -> pyarrow.schema:
+        glides_trip_key = pyarrow.struct(
+            [
+                ("serviceDate", pyarrow.string()),
+                ("startLocation", self.glides_location),
+                ("endLocation", self.glides_location),
+                ("startTime", pyarrow.string()),
+                ("endTime", pyarrow.string()),
+                ("revenue", pyarrow.string()),
+                ("glidesId", pyarrow.string()),
+            ]
+        )
+
+        glides_trip_update = pyarrow.struct(
+            [
+                ("previousTripKey", glides_trip_key),
+                ("type", pyarrow.string()),
+                ("tripKey", glides_trip_key),
+                ("comment", pyarrow.string()),
+                ("startLocation", self.glides_location),
+                ("endLocation", self.glides_location),
+                # can be "unset" string :(
+                ("startTime", pyarrow.string()),
+                # can be "unset" string :(
+                ("endTime", pyarrow.string()),
+                # an array of objects
+                ("cars", pyarrow.string()),
+                ("revenue", pyarrow.string()),
+                ("dropped", pyarrow.string()),
+                # an object with an array of objects
+                ("scheduled", pyarrow.string()),
+            ]
+        )
+
+        return pyarrow.schema(
+            [
+                (
+                    "data",
+                    pyarrow.struct(
+                        [
+                            ("metadata", self.glides_metadata),
+                            ("tripUpdates", pyarrow.list_(glides_trip_update)),
+                        ]
+                    ),
+                ),
+                ("id", pyarrow.string()),
+                ("type", pyarrow.string()),
+                ("time", pyarrow.timestamp("ms", tz="UTC")),
+                ("source", pyarrow.string()),
+                ("specversion", pyarrow.string()),
+                ("dataschema", pyarrow.string()),
+            ]
+        )
+
+    @property
+    def unique_key(self) -> str:
+        return "tripUpdates"
+
+    def convert_records(self) -> pd.Dataset:
+        def flatten_multitypes(record: Dict) -> Dict:
+            """
+            For each update in a record, flatten out the objects in "cars",
+            "dropped", and "cars". These objects are poorly structured for our
+            needs and can't be strongly typed. Leave them as json strings for
+            now and parse them later when analyzing.
+            """
+            for update in record["data"]["tripUpdates"]:
+                # convert these objects into strings if they exist
+                for key in ["scheduled", "dropped", "cars"]:
+                    try:
+                        update[key] = str(update[key])
+                    except KeyError:
+                        pass
+
+            return record
```

Comment:
i did this at the incoming record level as i couldn't figure out how to do it when calling `pyarrow.Table.from_pylist`